bundles/org.openhab.binding.dirigera/src/main/java/org/openhab/binding/dirigera/internal/handler/light/ColorLightHandler.java

```diff
@@ -0,0 +1,270 @@
+                if (targetChannel != null) {
+                    // apply and update to hsbCurrent, only in case !isOn deliver fake brightness HSBs
+                    switch (targetChannel) {
+                        case CHANNEL_LIGHT_COLOR:
+                            switch (key) {
+                                case "colorHue":
+                                    double hueValue = attributes.getInt(key);
+                                    hsbDevice = new HSBType(new DecimalType(hueValue), hsbDevice.getSaturation(),
+                                            hsbDevice.getBrightness());
+                                    hsbStateReflection = new HSBType(new DecimalType(hueValue),
+                                            hsbStateReflection.getSaturation(), hsbStateReflection.getBrightness());
+                                    deliverHSB = true;
+                                    break;
+                                case "colorSaturation":
+                                    double saturationValue = Math.round(attributes.getDouble(key) * 100);
+                                    hsbDevice = new HSBType(hsbDevice.getHue(), new PercentType((int) saturationValue),
+                                            hsbDevice.getBrightness());
+                                    hsbStateReflection = new HSBType(hsbStateReflection.getHue(),
+                                            new PercentType((int) saturationValue), hsbStateReflection.getBrightness());
+                                    deliverHSB = true;
+                                    break;
+                                case "lightLevel":
+                                    int brightnessValue = attributes.getInt(key);
+                                    // device needs the right values
+                                    hsbDevice = new HSBType(hsbDevice.getHue(), hsbDevice.getSaturation(),
+                                            new PercentType(brightnessValue));
+                                    hsbStateReflection = new HSBType(hsbStateReflection.getHue(),
+                                            hsbStateReflection.getSaturation(), new PercentType(brightnessValue));
+                                    deliverHSB = true;
+                                    break;
+                            }
+                            break;
+                    }
+                }
+            }
+            if (deliverHSB) {
+                updateState(new ChannelUID(thing.getUID(), CHANNEL_LIGHT_COLOR), hsbStateReflection);
+            }
+        }
+    }
+
+    /**
+     * Simulate color-temperature if color light doesn't have the "canReceive" "colorTemperature" capability
+     * https://www.npmjs.com/package/color-temperature?activeTab=code
+     *
+     * @param kelvin
+     * @return color temperature as HSBType
+     */
+    public static HSBType getHSBTemperature(long kelvin) {
```

Comment:
I'm no expert in color temperature. Maybe @andrewfg can share his thoughts. I wonmder if this can be replacde by core functionality

The core `ColorUtil` has some methods that can convert kelvin to HSB (`kelvinToXY` -> `xyToHsb`)
For reference:  https://www.openhab.org/javadoc/latest/org/openhab/core/util/colorutil

Reply:
> Maybe @andrewfg can share his thoughts. I wonmder if this can be replacde by core functionality

Indeed. You can use the the following `ColorUtil` methods..

- double kelvin = `ColorUtil.xyToKelvin(ColorUtil.hsbToXY(yourHsbType))`
- HSBType hsb = `ColorUtil.xyToHsb(ColorUtil.kelvinToXY(yourKelvinValue))`

Reply:
@weymann  can you proceed?

Reply:
> @weymann can you proceed?

Used this conversion but result is not satisfying.
With max temperature settings 
- bulb on the left with native temperature support is much warmer 
- bulb on the right without native temperature support with ColorUtil conversion looks cold
<img width="542" alt="image" src="https://github.com/user-attachments/assets/6ca9d310-4db2-4780-a569-053529782d2e" />


The conversion I found in [forum](https://community.openhab.org/t/colour-temp-to-hsb-conversion/32153) delivers nearly perfect result as visible result. Testing this ColorUtil and forum conversion are delivering different results in Hue and saturation.

![image](https://github.com/user-attachments/assets/e1683f40-3cb2-4d52-9590-adf295ca64a8)

Reply:
> With **max** temperature settings
> bulb on the right without native temperature support with ColorUtil conversion looks cold

I still think you are mixing things up. The **max** color temperature is SUPPOSED to be cold; and the **min** color temperature is supposed to be warm.

Reply:
Just for the avoidance of doubt..
- low color temperature in kelvin => cool in degrees => warmer color to the eye
- high color temperature in kelvin => hot in degrees => cooler color to the eye

Reply:
Understood and the code is already working this way.

In my comments I refer to the _values_ `ColorUtil` delivers.
Posted picture shows 
- left bulb with native color temperature support, it accepts kelvin values => 2200 kelvin 
- right bulb without native color temperature support  => 2200 kelvin converted with ColorUtil to HSB 36.5,68,100

At least for the IKEA system collors don't match. Does your HUE system shows the same if you switch from 2200 kelvin to HSB 36.5,68,100?

Reply:
> Does your HUE system shows the same if you switch from 2200 kelvin to HSB 36.5,68,100?

Yes.

![image](https://github.com/user-attachments/assets/de03d7c2-6428-4d65-8cf4-7b8f63297034)

Reply:
Ok, so ColorUtil is perfectly matching with Hue System xy values. But for IKEA these values doesn't match. 
This explains the pain when xy conversion is done for `kelvinToHSB` and the `closeTo` functions. 
So I'll stay on the current implemented solution.